server/Models/getRestaurantInfo.js

```diff
@@ -1,66 +1,74 @@
+const getRestaurantInfo = async (req, res) => {
   const { restaurantId } = req.params;
 
-  const { sort, keyword, star } = req.body;
+  const { sort, keywords, star } = req.body;
   let orderByStr = '';
   let selectStr = orderByStr;
   let sortDirection = '';
   let keywordStr = '';
   let starStr = '';
 
   if (sort === 'highest rating') {
-    orderByStr = 'CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4';
+    orderByStr = ', CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4  AS avg_overal_from_reviews';
     sortDirection = ' DESC,';
   }
 
   if (sort === 'lowest rating') {
-    orderByStr = 'CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4';
+    orderByStr = ', CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4 AS avg_overal_from_reviews';
     sortDirection = ' ASC,';
   }
 
-  if (keyword !== undefined) {
-    for (let i = 0; i < keyword.length; i ++) {
-      keywordStr += `AND reviews.reviewText LIKE '%${keyword[i]}%' `;
+  if (keywords.length !== 0) {
+    for (let i = 0; i < keywords.length; i ++) {
+      keywordStr += `AND reviews.reviewText LIKE '%${keywords[i]}%' `;
     }
 
-    keywordStr = keywordStr.slice(0, -1);
+    keywordStr = keywordStr.trim();
   }
 
   if (orderByStr !== '') {
-    selectStr = ', ' + orderByStr + ' AS avg_overal_from_reviews';
+    selectStr = 'avg_overal_from_reviews';
   }
 
-  if (star !== undefined) {
+  if (star !== 0) {
     starStr = `AND CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4 >= ${star} AND CAST((reviews.foodrating+reviews.servicerating+reviews.ambiencerating+reviews.valuerating) AS FLOAT)/4 < ${star + 1}`;
   }
 
   const query = {
     text: `
-      SELECT *${selectStr} FROM restaurants
+      SELECT *${orderByStr} FROM restaurants
         INNER JOIN reviews
           ON restaurants.id=reviews.restaurantid
         INNER JOIN users
           ON users.id=reviews.userid
         WHERE
-          restaurants.id=$1
+          restaurants.id=${restaurantId}
           ${keywordStr}
           ${starStr}
         ORDER BY
-          ${orderByStr} ${sortDirection}
+          ${selectStr} ${sortDirection}
           reviews.reviewdate DESC
         LIMIT 10;
     `,
-    values: [ restaurantId ],
   };
 
-  pg.query(query, (err, data) => {
-    if (err) {
-      console.log(err);
-      res.status(500).json(err);
+  return redis.get(query.text, async (err, result) => {
+    if (result) {
+      const jsonResult = JSON.parse(result);
+      return res.status(200).json(jsonResult.rows);
+
     } else {
-      res.status(200).json(data.rows);
+      try {
+        const data = await pg.query(query);
+        const stringfyData = JSON.stringify(data);
+        redis.setex(query.text, 1800, stringfyData);
```

Comment:
no need to make another variable, can just do
redis.setex(query.text, 1800, JSON.stringify(data))